xrpl/asyncio/wallet/wallet_generation.py

```diff
@@ -150,34 +154,40 @@ def process_faucet_host_url(input_url: str) -> str:
     return final_url
 
 
-def get_faucet_url(url: str, faucet_host: Optional[str] = None) -> str:
+def get_faucet_url(network_id: Optional[int], faucet_host: Optional[str] = None) -> str:
     """
-    Returns the URL of the faucet that should be used, based on whether the URL is from
-    a testnet or devnet client.
+    Returns the URL of the faucet that should be used, based on network_id and
+    faucet_host
 
     Args:
-        url: The URL that the client is using to access the ledger.
-        faucet_host: A custom host to use for funding a wallet.
+        network_id: The network_id corresponding to the XRPL network. This is parsed
+            from a ServerInfo() rippled response
+        faucet_host: A custom host to use for funding a wallet. This is useful because
+            network_id of sidechains might not be well-known. If faucet_url cannot be
+            gleaned from network_id alone, faucet_host is used
 
     Returns:
         The URL of the matching faucet.
 
     Raises:
-        XRPLFaucetException: if the provided URL is not for the testnet or devnet.
+        XRPLFaucetException: if the provided network_id does not correspond to testnet
+            or devnet.
     """
     if faucet_host is not None:
         return process_faucet_host_url(faucet_host)
-    if "altnet" in url or "testnet" in url:  # testnet
-        return _TEST_FAUCET_URL
-    if "sidechain-net2" in url:  # sidechain issuing chain devnet
-        raise XRPLFaucetException(
-            "Cannot fund an account on an issuing chain. Accounts must be created via "
-            "the bridge."
-        )
-    if "devnet" in url:  # devnet
-        return _DEV_FAUCET_URL
+
+    if network_id is not None:
+        if network_id == 1:
+            return _TEST_FAUCET_URL
+        elif network_id == 2:
+            return _DEV_FAUCET_URL
+        elif network_id == 0:
+            raise XRPLFaucetException("Cannot create faucet with a client on mainnet.")
+        elif network_id < 0:
+            raise XRPLFaucetException("network_id cannot be negative.")
```

Comment:
Does rippled allow this scenario?

Reply:
the config file does not allow negative numbers as input for network_id. But it accepts really large numbers (for eg: `4294967295` which could be negative numbers, if typecast into smaller signed-types)

also, rippled unit tests allow negative numbers - https://github.com/XRPLF/rippled/compare/develop...ckeshava:rippled:negNetworkID

Reply:
I don't think xrpl-py needs any specific checks for this, "we don't understand this network ID" is fine.

Reply:
updated in [f06e5c1](https://github.com/XRPLF/xrpl-py/pull/715/commits/f06e5c15e618816ee7478909695f37a38618de1d)